schemas/host.json

```diff
@@ -37,7 +37,7 @@
       "oneOf": [
         {
           "type": "string",
-          "pattern": "^(?:[01]\\d|2[0-3]):[0-5]\\d:[0-5]\\d$"
+          "pattern": "^(?:-1)$|^(?:[01]\\d|2[0-3]):[0-5]\\d:[0-5]\\d$"
```

Comment:
You could simplify to `^(-1)|(?:[01]\\d|2[0-3]):[0-5]\\d:[0-5]\\d$`

Reply:
The reason I added an extra "^" and "$" is so a string like "-1:00:00" wouldn't be accepted. Should this be a concern?

Reply:
Doesn’t my regex also prevent that?

Reply:
At least not when I tested it -- I believe it has to do with the parentheses grouping. The `|` chooses between `(-1)` and `(?:[01]\\d|2[0-3])` and then continues to match `:[0-5]\\d:[0-5]\\d`. I'm sure there's a better regex than the one I came up with. The only other one I could come up with was `^((?:-1)|((?:[01]\d|2[0-3]):[0-5]\d:[0-5]\d))$`.

`^(?:-1)|((?:[01]\d|2[0-3]):[0-5]\d:[0-5]\d)$` didn't quite behave right either because I think it chooses between `^(?:-1)` and `((?:[01]\d|2[0-3]):[0-5]\d:[0-5]\d)$` so the beginning and end of line markers get split

Reply:
Yes I had an mistake, I meant `^(-1|(?:[01]\d|2[0-3]):[0-5]\d:[0-5]\d)$` which I think works. the goal was to not have to redefine the begin/end line markers. If this works use it since it saves a few chars, otherwise just use your original :)

Reply:
👍 that works and is better